dnsapi/dns_duckdns.sh

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/bash
+
+# DNS API for DuckDNS
+#
+# Report bugs at https://github.com/RockyTV/acme.sh/issues
+#
+# Set this environment variable to match your DuckDNS account token:
+# DUCKDNS_TOKEN=aaaaaaaa-bbbb-cccc-dddddddddddd
+
+########  Public functions #####################
+# Usage: dns_duckdns_add   _acme-challenge.www.domain.com   "XKrxpRBosdIKFzxW_CT3KLZNf6q0HG9i01zxXp5CPBs"
+dns_duckdns_add() {
+  fulldomain=$1
+  txtvalue=$2
+
+  _info "Adding TXT record to DuckDNS domain"
+  _debug "fulldomain: $fulldomain"
+  _debug "txtvalue: $txtvalue"
+  
+  if [ -z "$DUCKDNS_TOKEN" ]; then
+    _err "DuckDNS token is not defined."
+    _err "Please export your account token to DUCKDNS_TOKEN and try again."
+    return 1
+  fi
+  _saveaccountconf DUCKDNS_TOKEN "$DUCKDNS_TOKEN"
+  
+  duckdns_url="https://www.duckdns.org/update?domains=$_domain&token=$DUCKDNS_TOKEN&txt=$txtvalue"
```

Comment:
It comes from acme.sh itself. A lot of DNS APIs use it, take a look: https://github.com/Neilpang/acme.sh/search?utf8=%E2%9C%93&q=_domain&type=